src/test/java/org/minions/devfund/reinaldo/WhackAMoleTest.java

```diff
@@ -0,0 +1,191 @@
+        Assert.assertTrue(whackAMole.place(0, 0));
+        Assert.assertFalse(whackAMole.place(0, 0));
+    }
+
+    /**
+     *Test score.
+     */
+    @Test
+    public void testWhackScore() {
+        whackAMole.place(0, 0);
+        whackAMole.place(1, 1);
+        whackAMole.place(2, 2);
+        whackAMole.whack(0, 0);
+        Assert.assertEquals(EXPECTED, whackAMole.getScore());
+    }
+
+    /**
+     *Test for AttemptsLeft.
+     */
+    @Test
+    public void testWhackAttemptsLeft() {
+        whackAMole.place(0, 0);
+        whackAMole.place(1, 1);
+        whackAMole.place(2, 2);
+        whackAMole.whack(0, 0);
+        Assert.assertEquals(EXPECTED1, whackAMole.getAttemptsLeft());
+    }
+
+    /**
+     *Test MolesLeft.
+     */
+    @Test
+    public void testWhackMolesLeft() {
+        whackAMole.place(0, 0);
+        whackAMole.place(1, 1);
+        whackAMole.place(2, 2);
+        whackAMole.whack(0, 0);
+        Assert.assertEquals(EXPECTED2, whackAMole.getMolesLeft());
+    }
+
+    /**
+     * Print the grid with M and W.
+     */
+    @Test
+    public void printGridTest() {
+        final int gridDimension = 3;
+        final int placementX = 0;
+        final int placementY = 1;
+        final int whackRow = 0;
+        final int whackColumn = 0;
+        final String expectedGrid = "W  M  *  \n*  *  *  \n*  *  *  \n";
+        whackAMole = new WhackAMole(NUM_ATTEMPTS, gridDimension);
+        whackAMole.whack(whackRow, whackColumn);
+        whackAMole.place(placementX, placementY);
+        assertEquals(expectedGrid, whackAMole.printGrid());
+    }
+
+    /**
+     * Print grid to user .
+     */
+    @Test
+    public void testPrintGridWhackToUser() {
+        final int gridDimension = 3;
+        final int whackRow = 0;
+        final int whackColumn = 0;
+        final String expectedGrid = "W  *  *  \n*  *  *  \n*  *  *  \n";
+        whackAMole = new WhackAMole(NUM_ATTEMPTS, gridDimension);
+        whackAMole.whack(whackRow, whackColumn);
+        assertEquals(expectedGrid, whackAMole.printGridToUser());
+    }
+
+    /**
+     * Print grid to user .
+     */
+    @Test
+    public void testChangeMbyCharacterGridWhackToUser() {
+        final int gridDimension = 3;
+        final int placementX = 0;
+        final int placementY = 1;
+        final int whackRow = 0;
+        final int whackColumn = 0;
+        final String expectedGrid = "W  *  *  \n*  *  *  \n*  *  *  \n";
+        whackAMole = new WhackAMole(NUM_ATTEMPTS, gridDimension);
+        whackAMole.whack(whackRow, whackColumn);
+        whackAMole.place(placementX, placementY);
+        assertEquals(expectedGrid, whackAMole.printGridToUser());
+    }
+
+    /**
+     * Mole on  a pleace.
+     */
+    @Test
+    public void testPutAMoleAPleace() {
+        final int gridDimension = 3;
+        final int placementX = 0;
+        final int placementY = 0;
+        WhackAMole whackAMole = new WhackAMole(NUM_ATTEMPTS, gridDimension);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "whackAMole" which hides the field declared at line 20. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)